pkg/jobmon/spool.go

```diff
@@ -95,14 +93,8 @@ func (s *SpoolManager) FinishJob(uniqID string, r *JobRun) error {
 }
 
 func (s *SpoolManager) GetFinishedJobs() ([]string, []*JobRun, error) {
-	err := s.getLock("")
```

Comment:
`s.getLock("")` was used as a lock for the jobs queue in order not to race between GetFinishedJobs() and RemoveJobs()